standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3252,7 +3252,7 @@ public Partition getPartitionWithAuth(String catName, String dbName, String tblN
       openTransaction();
       MPartition mpart = getMPartition(catName, dbName, tblName, partVals, null);
       if (mpart == null) {
-        commitTransaction();
+        success = commitTransaction();
```

Comment:
why not just move `commitTransaction()` to finally and remove the flag completely? this is a simple select, there is nothing to rollback

Reply:
Addressed.